sphinx/application.py

```diff
@@ -298,10 +299,18 @@ def create_builder(self, buildername):
         if buildername is None:
             logger.info(_('No builder selected, using default: html'))
             buildername = 'html'
-        if buildername not in self.builderclasses:
-            raise SphinxError(_('Builder name %s not registered') % buildername)
-
-        builderclass = self.builderclasses[buildername]
+        if buildername in self.builderclasses:
+            builderclass = self.builderclasses[buildername]
+        else:
+            entry_points = iter_entry_points('sphinx_builders', buildername)
+            try:
+                entry_point = next(entry_points)
+            except StopIteration:
+                raise SphinxError('Builder name %s not registered or available'
+                                  ' through entry point' % buildername)
+            else:
+                builderclass = entry_point.load()
+                load_extension(self, builderclass.__module__)
```

Comment:
`builderclass.__module__` is a string as well, so it's equivalent. But I missed the `EntryPoint.module_name`  attribute before and I agree that it is better to use it instead.

I'm was not sure about your suggestion to not make use of `EntryPoint.load()`. It does fit in better with the existing way of handling extensions, but it seems a bit strange to only use the entry point for the module name. But it turns out that an entry point can be just the module:

> [setuptools docs](https://setuptools.readthedocs.io/en/latest/setuptools.html#dynamic-discovery-of-services-and-plugins): An entry point specifier consists of a name and value, separated by an = sign. The value consists of a dotted module name, **optionally** followed by a : and a dotted identifier naming an object within the module.

`entry_point.load()` then returns the module itself (not the name). So it still doesn't make sense loading the entry point, but I do think this is an improvement since it fits in better with the existing extension handling. Additionally, the builder class now doesn't have to be be defined in the extension module itself (for whatever reason).

PR updated.